libensemble/alloc_funcs/give_pregenerated_work.py

```diff
@@ -14,11 +14,15 @@ def give_pregenerated_sim_work(W, H, sim_specs, gen_specs, alloc_specs, persis_i
     if not persis_info:
         persis_info['next_to_give'] = 0
 
-    assert persis_info['next_to_give'] < len(H), 'No more work to give inside give_pregenerated_sim_work.'
+    if persis_info['next_to_give'] >= len(H):
+        return Work, persis_info, 1
 
     for i in avail_worker_ids(W):
         # Give sim work
         sim_work(Work, i, sim_specs['in'], [persis_info['next_to_give']], [])
```

Comment:
Should it check 'given' for the row its on - and increment if True?

Reply:
Yes. That is possible. This allocation function is called `give_pregenerated_sample` so I think that it's okay to just go through all of the rows of `H` in order. (But I'm fine with checking `given`.)

Note that the top of this allocation function is not very robust: If there is `persis_info` then `next_to_give` will not be initialized. I hope that such issues will be addressed in our re-working of allocation functions.

Reply:
Should it just be checking the 'next_to_give' component is initialized? Something like:

    persis_info['next_to_give'] = persis_info.get('next_to_give', 0) + 1

Reply:
The initialization is fixed in 8080b1b4

Reply:
Thank you for the feedback!